### PR TITLE
Add readUser for RoomEvent

### PR DESCRIPTION
**Pull request by encombhat**

### CODE REVIEW

**Comment by KitsuneRal** [comment]
The fact that `*(q->findInTimeline())->` looks pretty ugly speaks for itself :) The list of users should be added not to `RoomEvent` but rather to `TimelineItem`; alternatively, you might even just add a hashmap from event ids to users to `Room::Private`; then instead of `findInTimeline` you'd just need a lookup in the hashmap.

**Comment by encombhat** [comment]
I added a list in Room::Private. 
(It seems that I force pushed)

**Comment by KitsuneRal** [comment]
Great! And to simplify further, you could use `QMultiHash` (sorry, just occurred to me).

**Comment by encombhat** [comment]
It seems working when I tested it. :)

File: lib/room.cpp [hunk 1]
```diff
@@ -641,7 +641,7 @@ QString Room::readMarkerEventId() const
 
 QList<User*> Room::usersAtEventId(const QString& eventId) {
     if (!d->eventIdReadUsers.contains(eventId)) return QList<User*>();
```
Comment (KitsuneRal) [inline lib/room.cpp]: You don't need to special-case here: `values()` will do the right thing.

**Comment by KitsuneRal** [comment]
One last nitpick, and I'm merging it.

**Comment by encombhat** [comment]
There is an issue when testing.
A lot of users are not showing up in the read receipt for an event if their read receipt does not change. I guess it needs to be fetched manually.

**Comment by encombhat** [comment]
Wait a sec, it seems that it need an extra signal.

**Comment by KitsuneRal** [comment]
The whole thing looks pretty good. What's up with those missing read receipts you mentioned in the room, have you found them?

**Comment by encombhat** [comment]
I still can't find them. Guess it is impossible to fetch all m.receipt in all rooms in one /sync.
Let me check some documents.

**Comment by encombhat** [comment]
In nheko it is done using (persistent)Cache: https://github.com/mujx/nheko/blob/183975be1aeef9965a179eb1159d66e818b00ddd/src/Cache.cpp#L178

**Comment by encombhat** [comment]
Any progress of this?

**Comment by KitsuneRal** [comment]
Sorry, got entangled in too many other things. Thanks for submitting!

**Pull request merged**